src/compas_eve/core.py

```diff
@@ -43,6 +45,7 @@ def id_counter(self):
         return self._id_counter
 
     def publish(self, topic, message):
+        self.codec.encode(message)
         pass
```

Comment:
This pass has no effect now, should be removed. But perhaps more importantly, this is probably not where we should be calling the encoding. This method is empty because the implementation is on the sub-classes of transport. So, I would leave it empty.